__tests__/utils/permissions.test.js

```diff
@@ -48,27 +56,21 @@ describe('requestExternalStoragePermission shpuld run correctly', () => {
     }));
   });
   it('requestExternalStoragePermission granted should run correctly', async () => {
-    jest.spyOn(mock, 'check').mockResolvedValue('granted');
+    jest.spyOn(mock, 'requestMultiple').mockResolvedValue(mockPermission('granted'));
     const checkPermission = await requestExternalStoragePermission();
     expect(mock.check).toHaveBeenCalled();
     expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
   });
   it('requestExternalStoragePermission unavailable should run correctly', async () => {
-    jest.spyOn(mock, 'check').mockResolvedValue('unavailable');
+    jest.spyOn(mock, 'requestMultiple').mockResolvedValue(mockPermission('unavailable'));
     const checkPermission = await requestExternalStoragePermission();
     expect(mock.check).toHaveBeenCalled();
-    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionUnavailable);
+    expect(checkPermission.message).toEqual(StringConstant.externalStoragePermissionUnavailable);
   });
   it('requestExternalStoragePermission denied should run correctly', async () => {
-    jest.spyOn(mock, 'check').mockResolvedValue('denied');
-    const checkPermission = await requestExternalStoragePermission();
-    expect(mock.check).toHaveBeenCalled();
-    expect(checkPermission.message).toEqual(StringConstant.cameraPermissionGranted);
-  });
-  it('requestExternalStoragePermission default should run correctly', async () => {
-    jest.spyOn(mock, 'check').mockResolvedValue('default');
+    jest.spyOn(mock, 'requestMultiple').mockResolvedValue(mockPermission('denied'));
     const checkPermission = await requestExternalStoragePermission();
     expect(mock.check).toHaveBeenCalled();
-    expect(checkPermission.message).toEqual('');
+    expect(checkPermission.message).toEqual(StringConstant.externalStoragePermissionDenied);
   });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The tests are mocking the `requestMultiple` function and checking if the `check` function has been called. However, there is no call to the `check` function in the new code. This could lead to false positives in your tests. You should either mock the correct function or change the test to reflect the actual behavior of the code.
```diff
-     expect(mock.check).toHaveBeenCalled();
+     expect(mock.requestMultiple).toHaveBeenCalled();
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->